packages/endpoints-overlay/src/endpoint.js

```diff
@@ -18,7 +18,31 @@ import {
 
 import * as Styled from "./styled";
 
-function Icon({ type }) {
+function DefaultMapMarkerIcon({ location, type }) {
+  return (
+    <Styled.StackedIconContainer title={location.name}>
+      {type === "from" ? (
+        // From icon should have white circle background
+        <>
+          <Styled.StackedCircle size={24} />
+          <Styled.StackedLocationIcon size={24} type={type} />
+        </>
+      ) : (
```

Comment:
These are really two different icons.

Reply:
I don't think this deserves a refactor as suggested. I think the single custom component prop as currently designed provides an additional benefit of leaving more things open to the implementing application by providing both the `location` and `type` properties. In this way the implementing app is unconstrained and could implement more than just 2 icon types perhaps based on location attributes or even use a single icon type.

Reply:
If multiple icons can be returned by the implementer, then we should rename `MapMarkerIcon` to `mapMarkerIconProvider`.

Reply:
At least, rename MapMarkerIcon to something that suggests that different icons can/will be rendered.